File: test/stack/valid-number.test.ts
import { isNumber } from '../../src/stack/valid-number';

describe('valid number', () => {
  test('valid number - test case 1', async () => {
    expect(isNumber('2')).toBe(true);
  });

  test('valid number - test case 2', async () => {
    expect(isNumber('0089')).toBe(true);
  });

  test('valid number - test case 3', async () => {
    expect(isNumber('-0.1')).toBe(true);
  });

  test('valid number - test case 4', async () => {
    expect(isNumber('+3.14')).toBe(true);
  });

  test('valid number - test case 5', async () => {
    expect(isNumber('4.')).toBe(true);
  });

  test('valid number - test case 6', async () => {
    expect(isNumber('-.9')).toBe(true);
  });

  test('valid number - test case 7', async () => {
    expect(isNumber('2e10')).toBe(true);
  });

  test('valid number - test case 8', async () => {
    expect(isNumber('-90E3')).toBe(true);
  });

  test('valid number - test case 9', async () => {
    expect(isNumber('3e+7')).toBe(true);
  });

  test('valid number - test case 10', async () => {
    expect(isNumber('+6e-1')).toBe(true);
  });

  test('valid number - test case 11', async () => {
    expect(isNumber('53.5e93')).toBe(true);
  });

  test('valid number - test case 12', async () => {
    expect(isNumber('-123.456e789')).toBe(true);
  });

  test('valid number - test case 13', async () => {
    expect(isNumber('46.e3')).toBe(true);
  });

  test('valid number - test case 14', async () => {
    expect(isNumber('1.e10')).toBe(true);
  });

  test('valid number - test case 15', async () => {
    expect(isNumber('9.e3')).toBe(true);
  });

  test('valid number - test case 16', async () => {
    expect(isNumber('abc')).toBe(false);
  });

  test('valid number - test case 17', async () => {
    expect(isNumber('1a')).toBe(false);
  });

  test('valid number - test case 18', async () => {
    expect(isNumber('1e')).toBe(false);
  });

  test('valid number - test case 19', async () => {
    expect(isNumber('e3')).toBe(false);
  });

  test('valid number - test case 20', async () => {
    expect(isNumber('99e2.5')).toBe(false);
  });

  test('valid number - test case 21', async () => {
    expect(isNumber('--6')).toBe(false);
  });

  test('valid number - test case 22', async () => {
    expect(isNumber('-+3')).toBe(false);
  });

  test('valid number - test case 23', async () => {
    expect(isNumber('95a54e53')).toBe(false);
  });

  test('valid number - test case 24', async () => {
    expect(isNumber('1ee3')).toBe(false);
  });

  test('valid number - test case 25', async () => {
    expect(isNumber('.')).toBe(false);
  });

  test('valid number - test case 26', async () => {
    expect(isNumber('1e+')).toBe(false);
  });

  test('valid number - test case 27', async () => {
    expect(isNumber('1e+.')).toBe(false);
  });

  test('valid number - test case 28', async () => {
    expect(isNumber('+.')).toBe(false);
  });

  test('valid number - test case 29', async () => {
    expect(isNumber('.+')).toBe(false);
  });

  test('valid number - test case 30', async () => {
    expect(isNumber('+e3')).toBe(false);
  });

  test('valid number - test case 31', async () => {
    expect(isNumber('46e.3')).toBe(false);
  });

  test('valid number - test case 32', async () => {
    expect(isNumber('.e3')).toBe(false);
  });
});

File: test/leetcode/stack/valid_number_test.py
from leetcode.stack.valid_number import Solution


soln = Solution()


def test_case_1():
    assert soln.isNumber("2")


def test_case_2():
    assert soln.isNumber("0089")


def test_case_3():
    assert soln.isNumber("-0.1")


def test_case_4():
    assert soln.isNumber("+3.14")


def test_case_5():
    assert soln.isNumber("4.")


def test_case_6():
    assert soln.isNumber("-.9")


def test_case_7():
    assert soln.isNumber("2e10")


def test_case_8():
    assert soln.isNumber("-90E3")


def test_case_9():
    assert soln.isNumber("3e+7")


def test_case_10():
    assert soln.isNumber("+6e-1")


def test_case_11():
    assert soln.isNumber("53.5e93")


def test_case_12():
    assert soln.isNumber("-123.456e789")


def test_case_13():
    assert soln.isNumber("46.e3")


def test_case_14():
    assert soln.isNumber("1.e10")


def test_case_15():
    assert soln.isNumber("9.e3")


def test_case_16():
    assert not soln.isNumber("abc")


def test_case_17():
    assert not soln.isNumber("1a")


def test_case_18():
    assert not soln.isNumber("1e")


def test_case_19():
    assert not soln.isNumber("e3")


def test_case_20():
    assert not soln.isNumber("99e2.5")


def test_case_21():
    assert not soln.isNumber("--6")


def test_case_22():
    assert not soln.isNumber("-+3")


def test_case_23():
    assert not soln.isNumber("95a54e53")


def test_case_24():
    assert not soln.isNumber("1ee3")


def test_case_25():
    assert not soln.isNumber(".")


def test_case_26():
    assert not soln.isNumber("1e+")


def test_case_27():
    assert not soln.isNumber("1e+.")


def test_case_28():
    assert not soln.isNumber("+.")


def test_case_29():
    assert not soln.isNumber(".+")


def test_case_30():
    assert not soln.isNumber("+e3")


def test_case_31():
    assert not soln.isNumber("46e.3")


def test_case_32():
    assert not soln.isNumber(".e3")
